### PR TITLE
Bug 1215654 - [Metrics] Metrics may not be recorded when level set to…

### PR DESCRIPTION
**Pull request by russnicoletti**
… enhanced

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=65ac4410e96e25fa0e354995c68fa7997f9ac6db

File: apps/settings/js/panels/improve_browser_os/panel.js [hunk 1]
```diff
@@ -93,11 +96,34 @@ define(function(require) {
...
         level = LEVELSMAP[level];
         if (level) {
-          var metricLevel = {};
-          metricLevel[METRICLEVELSETTING] = level;
-          navigator.mozSettings.createLock().set(metricLevel);
+          updateSetting(METRICLEVELSETTING, level);
+
+          // If the metrics level is being set to 'enhanced', DevTools
+          // must enabled. Also, the HUD widgets should be hidden.
+          if (level === 'Enhanced') {
```
Comment (gasolin) [inline apps/settings/js/panels/improve_browser_os/panel.js]: use `=== LEVELSMAP['Enhanced']` so it wont introduce any case error

File: apps/settings/js/panels/improve_browser_os/panel.js [hunk 2]
```diff
@@ -93,11 +96,34 @@ define(function(require) {
       _setMetricLevel: function(level) {
         level = LEVELSMAP[level];
         if (level) {
-          var metricLevel = {};
-          metricLevel[METRICLEVELSETTING] = level;
-          navigator.mozSettings.createLock().set(metricLevel);
+          updateSetting(METRICLEVELSETTING, level);
+
+          // If the metrics level is being set to 'enhanced', DevTools
+          // must enabled. Also, the HUD widgets should be hidden.
```
Comment (gasolin) [inline apps/settings/js/panels/improve_browser_os/panel.js]: DevTools must be enabled to collect telemetry metrics.

File: apps/settings/js/panels/improve_browser_os/panel.js [hunk 3]
```diff
@@ -7,6 +7,9 @@ define(function(require) {
   const METRICLEVELSETTING = 'metrics.selectedMetrics.level';
   const DOGFOODSETTING = 'debug.performance_data.dogfooding';
   const SHAREDATASETTING = 'debug.performance_data.shared';
+  const DEVTOOLSOVERLAY = 'devtools.overlay';
+  const HIDEHUDWIDGETS = 'hud.hide';
```
Comment (gasolin) [inline apps/settings/js/panels/improve_browser_os/panel.js]: show/hide HUD widgets does not related to this function, we could just change its default value in common-settings.js

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3ef03d7d9f6c141a0a1a24e3c810b4fc7f482da6

File: apps/settings/js/panels/improve_browser_os/panel.js [hunk 4]
```diff
@@ -93,11 +95,30 @@ define(function(require) {
...
         level = LEVELSMAP[level];
         if (level) {
-          var metricLevel = {};
-          metricLevel[METRICLEVELSETTING] = level;
-          navigator.mozSettings.createLock().set(metricLevel);
+          updateSetting(METRICLEVELSETTING, level);
+
+          // If the metrics level is being set to 'enhanced', DevTools
+          // must enabled.
+          if (level === 'Enhanced') {
```
Comment (gasolin) [inline apps/settings/js/panels/improve_browser_os/panel.js]: use === LEVELSMAP['Enhanced'] so it wont introduce any case error

File: apps/settings/js/panels/improve_browser_os/panel.js [hunk 5]
```diff
@@ -93,11 +95,30 @@ define(function(require) {
       _setMetricLevel: function(level) {
         level = LEVELSMAP[level];
         if (level) {
-          var metricLevel = {};
-          metricLevel[METRICLEVELSETTING] = level;
-          navigator.mozSettings.createLock().set(metricLevel);
+          updateSetting(METRICLEVELSETTING, level);
+
+          // If the metrics level is being set to 'enhanced', DevTools
+          // must enabled.
```
Comment (gasolin) [inline apps/settings/js/panels/improve_browser_os/panel.js]: add why: `DevTools must be enabled to collect telemetry metrics.`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a2a62a8ad18055a6d7023b895a692b04345c061c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5b462d046c1b572f754dca38648965e93a6cb63f

**Pull request merged**